example/src/OutOfBandAssets.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { SafeAreaView, ScrollView, StyleSheet, Text } from 'react-native';
+import Rive, { Fit, RNRiveError } from 'rive-react-native';
+
+export default function StateMachine() {
+  return (
+    <SafeAreaView style={styles.safeAreaViewContainer}>
+      <ScrollView contentContainerStyle={styles.container}>
+        <Rive
+          autoplay={true}
+          fit={Fit.Contain}
+          style={styles.box}
+          stateMachineName="State Machine 1"
+          // You can use the `assetsHandled` prop to load in external assets from a URI
+          // or bundled asset on the native platform (iOS and Android)
+          // or as a source loaded directly from JavaScript.
+          //
+          // Below demonstrates various ways to load in the same asset
+          // located in different places. Its not needed to store the same
+          // asset in all these locations, but this example does that for
+          // demonstration purposes.
+          //
+          // The key of the map is the unique asset identifier (as exported in the Editor),
+          // which is a combination of the asset name and its unique identifier.
+          assetsHandled={{
+            'Inter-594377': {
+              source: require('./assets/Inter-594377.ttf'),
+              // source: {
+              //   fileName: 'Feather.ttf',
+              //   path: 'fonts', // only needed for Android assets
+              // },
+            },
+            'referenced-image-2929282': {
+              source: require('./assets/referenced-image-2929282.png'),
+              // source: {
+              //   uri: 'https://picsum.photos/id/270/500/500',
+              // },
+              // source: {
+              //   fileName: 'referenced-image-2929282.png',
+              //   path: 'images', // only needed for Android assets
+              // },
+            },
+            'referenced_audio-2929340': {
```

Comment:
There's no scenario where the asset IDs might change, is there? For example, if the Rive file gets copied or if you reimport a PSD? That could be painful if you have a giant object with ID's included.

Reply:
That's why the approach for the other runtimes are a callback mechanism where you can read the name + id and query the file. That's not possible for React Native at this stage, which is why this API is different.

The best way to circumvent this is to provide a "name" only fallback and forgo the id